app/_hub/kong-inc/opa/0.1.x.md

```diff
@@ -0,0 +1,314 @@
+      required: false
+      datatype: boolean
+      default: false
+      description: |
+        If set to true, the Kong Gateway Consumer object in use for the current request (if any) is included as input to OPA.
+---
+
+## Usage
+
+### Set up an OPA server
+
+If you are not already running an OPA server, start one by following instructions
+on [openpolicyagent.org](https://openpolicyagent.org).
+
+### Set up a dummy policy
+
+Create an `example.rego` file with the following content:
+
+```rego
+package example
+
+default allow = false
+
+allowBoolean {
+  header_present
+}
+
+header_present {
+  input.request.http.headers["my-secret-header"] == "open-sesame"
+}
+
+allowDetailed = response {
+  header_present
+  response := {
+    "allow": true,
+    "headers": {
+      "header-from-opa": "accepted",
+    },
+  }
+}
+
+allowDetailed = response {
+  not header_present
+  response := {
+    "allow": false,
+    "status": 418,
+    "headers": {
+      "header-from-opa": "rejected",
+    },
+  }
+}
+```
+
+This policy allows the request to pass through if it contains the HTTP header
+`my-secret-header` with value `open-sesame`.
+
+
+Next, configure the policy with OPA:
+
+```bash
+curl -XPUT localhost:8181/v1/policies/example --data-binary @example.rego
+```
+
+The above command uses OPA's default port 8181. It could be different for your
+setup.
+
+### Set up Kong Gateway
+
+Set up a Route and Service in {{site.base_gateway}} and then enable the plugin:
+
+```bash
+curl -X POST http://<admin-hostname>:8001/routes/<route>/plugins \
+    --data "name=opa"  \
+    --data "config.opa_path=/v1/data/example/allowBoolean" \
+    --data "config.opa_host=<host-where-opa is running>"
+```
+
+### Make a request
+
+Now, make a request to {{site.base_gateway}}:
+
+```bash
+curl http://kong:8000/{proxy_path}
+```
+
+You will get a 403 response from the gateway because OPA has rejected the request.
+
+Next, make the same request providing the necessary header:
+
+```bash
+curl http://kong:8000/{proxy_path} \
+  -H 'my-secret-header: open-sesame'
+```
+
+This time you will get the response from your upstream service, as this request
+is allowed by OPA.
+
+Next, update the plugin configuration to use `/v1/data/example/allowDetailed` as
+`opa_path` and observe how the policy affects the response and proxied request.
+
+## OPA input structure
+
+The input to OPA has the following JSON structure:
+
+```
+{
+  "input": {
+    "request": { # details about the request from client to Kong
+      "http": {
+        "host": "example.org", # host header used by the client to make the request
+        "port": "8000",        # port to which the request was made
+        "tls": {},             # TLS details if the request was made on HTTPS and Kong terminated the TLS connection
+        "method": "GET",       # HTTP method used in the request
+        "scheme": "http",      # protocol used to make the request by the client, this can be either `http` or `https`
```

Comment:
This was the enhancement in the new version:

```suggestion
        "scheme": "http",      # protocol used to make the request by the client, this can be either `http` or `https`
        "path": "/foo/bar",      # HTTP path used in the request
```